File: dotnet/Lexy/Lexy.Compiler/Language/Expressions/Functions/HoursFunction.cs
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Expressions.Functions
{
    public class HoursFunction : EndStartDateFunction
    {
        public const string Name = "HOURS";

        protected override string FunctionName => Name;

        private HoursFunction(Expression endDateExpression, Expression startDateExpression, SourceReference reference)
            : base(endDateExpression, startDateExpression, reference)
        {
        }

        public static BuiltInFunction Create(SourceReference reference, Expression endDateExpression,
            Expression startDateExpression) =>
            new HoursFunction(endDateExpression, startDateExpression, reference);
    }
}

File: typescipt/src/language/expressions/functions/hoursFunction.ts


export class HoursFunction extends EndStartDateFunction {
   public const string Name = `HOURS`;

   protected override string FunctionName => Name;

   constructor(endDateExpression: Expression, startDateExpression: Expression, reference: SourceReference)
     : base(endDateExpression, startDateExpression, reference) {
   }

   public static ExpressionFunction Create(SourceReference reference, Expression endDateExpression,
     Expression startDateExpression) {
     return new HoursFunction(endDateExpression, startDateExpression, reference);
   }
}
